### ISSUE TITLE
Collapsible TO, CC, BCC, and FROM Fields

### ISSUE THREAD

**Issue by GretaD** [nextcloud/mail#10421]
### Is your feature request related to a problem? Please describe.

When composing an email with recipients added in the TO, CC, BCC, and FROM fields, alongside attachments, the available space for the email body becomes significantly small. To improve the user experience, I propose making these fields collapsible, allowing users to expand or collapse them as needed.

The problem:
In scenarios where multiple recipients are added across all fields (TO, CC, BCC, and FROM) and attachments are included, the input space for the email body is reduced, making it difficult to focus on writing the message. 

Open for any other ideas @jancborchardt @nimishavijay @marcoambrosini 

### Describe the solution you'd like

Have a collapsible UI for the TO, CC, BCC, and FROM fields.
Users should be able to collapse these fields after entering the recipients, freeing up space for the body section.
The fields can be expanded again if the user needs to edit or review them.

### Additional context

![Image](https://github.com/user-attachments/assets/33bcc6e8-e338-4844-82d9-7b81fa3516d4)

**Comment by SebastianKrupinski** [comment]
Few ideas for a fix:
- Add ability to scroll to the modal
- Add ability to maximize the modal to the full screen 
- Collapse fields to a icon when not in focus, with a hover over list to emails

**Comment by GretaD** [comment]
> Few ideas for a fix:
> 
> * Add ability to scroll to the modal

We already scroll the body part, but doesnt fix the issue, because you have to read one line at the time and its very hard to see the text you are writing or if you want to read the full text you wrote before sending.

> * Add ability to maximize the modal to the full screen

yes, good point, but the issue is the height and on small screen probably its not gonna make e big difference. 

> * Collapse fields to a icon when not in focus, with a hover over list to emails

👍

**Comment by ChristophWurst** [comment]
I think Sebastian means to scroll the *whole modal*, which sounds like a good idea in general when we run out of space.

**Comment by SebastianKrupinski** [comment]
> I think Sebastian means to scroll the _whole modal_, which sounds like a good idea in general when we run out of space.

Correct, this is what I meant.

**Comment by thstyl2000** [comment]
Nice! Just wanted to post a new feature request :) Here's a mockup that in my opinion declutters and provides with more space:

![Image](https://github.com/user-attachments/assets/e7def222-820a-4f51-a97e-b47b6e794a34)

It will also help, if no email addresses are shown if there is a name, and only show the email addresses on hover.

Scrolling the whole modal is also a good idea, maybe just leave the subject line hanging on top when scrolling down.

**Comment by thstyl2000** [comment]
Looks great, thanks for that! A couple of suggestions:

- since when adding cc one doesn't always add bcc (and the other way around), what about making the cc and bcc fields separately collapsible?
- what about changing the "add contact or email address" with just a "+" button that could also go directly next to the last address and save a new line in the field (and to my impression declutter the ui even more)?
- one could also show only names if there is a name attached to the email, not only when it is a contact, ex replying to someone who is not on your contacts does not directly put him into some "recently contacted" contact list, right...?

**Comment by SebastianKrupinski** [comment]
@nimishavijay 

The input fields really only need to be visible when a user in adding a recipient, after that they are just wasted space. So...  

What do you think about having only, word or image buttons across the top, for "From, To, Cc, Bcc"? The buttons could be colorized depending on if the field has contents or not (grey for empty, blue for there are recipients). Clicking one of the buttons would show the full input field, then when the input is no longer in focus it collapses again to a button. Alternatively we could add a collapse icon at the end of the field, and let the user control the visibility. We could also couple this with a mouse over function that just shows the recipients, for cases where you want to double check the list of recipients.

I would do a mock up but don't have the right tools for this at the moment, so I hold my explanation is clear.

**Comment by nimishavijay** [comment]
@SebastianKrupinski 


> The input fields really only need to be visible when a user is adding a recipient, after that they are just wasted space 

Agreed, that was the idea behind the to/cc/bcc fields automatically collapsing into a single line when not focused :)

> What do you think about having only, word or image buttons across the top, for "From, To, Cc, Bcc"? 

This would indeed save some space in the composer modal, but it seems like a strong deviation from most other mail clients which always show the from and to fields at least. 

> We could also couple this with a mouse over function that just shows the recipients, for cases where you want to double check the list of recipients

That actually could be a good idea here too! On hovering over the "+2 more" or "3 in cc" etc the recipients could be shown in a tooltip.

@thstyl2000 

> since when adding cc one doesn't always add bcc (and the other way around), what about making the cc and bcc fields separately collapsible?

I would stick with one toggle, just to keep it simple :) so it's one click in the To field to see all the recipients and one click in subject/body to hide them 

> what about changing the "add contact or email address" with just a "+" button that could also go directly next to the last address and save a new line in the field (and to my impression declutter the ui even more)?

Honestly we could also do away with showing the "add contact..." placeholder altogether if the field is filled, as I'd assume once an email address has been added it's easy to add more. This would also keep it in line with the current behavior of the generic NcSelect component. What do you think @SebastianKrupinski ?

**Comment by TheNomad11** [comment]
Any news on this?

**Comment by schmidtinger** [comment]
Hello everyone,

Are there any updates regarding the fix for this bug following the release of the latest versions?

What is the current development status concerning the issue with the missing virtual keyboard/cursor in the mobile input fields?

Thank you for any information!

Best regards, Schmidtinger
